rollup/pipeline/pipeline.go

```diff
@@ -308,15 +308,20 @@ func (p *Pipeline) cccStage(candidates <-chan *BlockCandidate, deadline time.Tim
 	var deadlineReached bool
 
 	go func() {
+		deadlineTimer := time.NewTimer(time.Until(deadline))
 		defer func() {
 			close(resultCh)
+			if !deadlineReached && !deadlineTimer.Stop() {
+				<-deadlineTimer.C
+			}
 			lifetimeTimer.UpdateSince(p.start)
 		}()
 		for {
 			idleStart := time.Now()
 			select {
-			case <-time.After(time.Until(deadline)):
```

Comment:
nice catch. so a slow mem leak here previously? if it's not triggered.

Reply:
Yeah, it is not that significant in our case and it isn't leaked permantently but it adds unneccsary pressure to GC.

Reply:
got it. another thing is that previously if it cannot stop early, it will wait for another hour, but now it only waits for another clique period. (I think it's also feasible).